File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

bool compare(string s1, string s2){
    return s1 + s2 > s2 + s1;
}

string solution(vector<int> numbers) {
    vector<string> v;
    
    for(auto n : numbers) v.push_back(to_string(n));
    
    sort(v.begin(), v.end(), compare);
    if(v.at(0) == "0") return "0";
    string s;
    for(auto V : v) s += V;
    return s;
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;
import java.io.*;

class Solution {
    public String solution(int[] numbers) {
        
        List<String> str = new ArrayList<>();
        for(int number : numbers) {
            str.add(String.valueOf(number));
        }
        
        str.sort((a, b) -> {
            return (b+a).compareTo(a+b);
        });
        
        if(str.getFirst().equals("0")) return "0";
        
        StringBuilder sb = new StringBuilder();
        for(int i=0; i<str.size(); i++) {
            sb.append(str.get(i));
        }
        return sb.toString();
    }
}